src/modules/get-chip-items.ts

```diff
@@ -0,0 +1,61 @@
+import i18n from '@dhis2/d2-i18n'
+import type { SupportedAxis } from '@constants/axis-types'
+import type { SupportedDimensionType } from '@constants/dimension-types'
+import type { SupportedInputType } from '@constants/input-types'
+import type { SupportedValueType } from '@constants/value-types'
+
+const DIMENSION_ID_ORGUNIT = 'ou'
+
+interface ChipDimension {
+    id: string
+    dimensionType?: SupportedDimensionType
+    optionSet?: string
+    valueType?: SupportedValueType
+}
+
+interface GetChipItemsParams {
+    dimension: ChipDimension
+    conditionsLength: number | undefined
+    itemsLength: number | undefined
+    inputType: SupportedInputType
+    axisId: SupportedAxis
+}
+
+export const getChipItems = ({
+    dimension,
+    conditionsLength,
+    itemsLength,
+    inputType,
+    axisId,
+}: GetChipItemsParams): string | number | null => {
```

Comment:
This seems a little funky to me. The function name isn't very descriptive and the fact that it returns `string | number | null` suggests to me that perhaps this should be refactored a bit? 
I was thinking of splitting this up into 2 helpers:
1. `hasChipItems` returns false for cases where the current function returns `null`
2. `getChipItemsCount` return a number or `null` (can also call `hasChipItems` and return null if that is true, but that would be for tests only)

And then in the component we could:
- use `hasChipItems` for conditionally rendering the component
- render the content using `getChipItemsCount() ?? i18n.t('All')`

Reply:
This is really tricky code to refactor since there are a lot of weird "combos" that it checks. I'm taking this directly from LL. 

What if I instead always return a string? e.g. 'all', '2', ''. I know this doesn't solve the complexity of the function, but it at least cleans up the typing.

I'll implement that and get your feedback.

Reply:
Update. I did try to clean up the code somewhat, relying on the unit tests for help.